core/src/main/java/org/fao/geonet/api/records/attachments/StoreUtils.java

```diff
@@ -40,18 +40,20 @@
      * @param context
      * @param oldMetadataId               The source metadata ID
      * @param newMetadataId               The destination metadata ID
+     * @param oldApproved                 Old approved flag to use for the source.
+     * @param newApproved                 New approved flag to use on the destination.
```

Comment:
I would add some additional documentation explaining the usage of these parameters.

Seem `oldApproved` is always call with the value `true`, are there any cases where can be relevant to use the value `false`, if not maybe it can be removed and use only `newApproved`.

Reply:
In the existing code I don't believe there are cases for old approved being false.

The only possible case I can think of at the moment is if we were to add logic for duplicating a working copy (i.e a user may want to make a new metadata copy of their working copy before adding more changes.   In that case I believe the old approved would be false and the new approved would be true. But then the UUID would also need to change on the destination. I'm not sure if that case will ever exist?

I will remove the oldApproved and update the documentation as you suggested.